src/cargo/util/semver_eval_ext.rs

```diff
@@ -0,0 +1,520 @@
+use semver::{Comparator, Op, Prerelease, Version, VersionReq};
```

Comment:
Should we have some mod-level comments explaining what have been copied over from `semver` and why?

Reply:
No problems.